torch/testing/_internal/common_optimizers.py

```diff
@@ -405,6 +403,9 @@ def optim_inputs_func_adam(device, dtype=None):
     total = [
         OptimizerInput(params=None, kwargs={}, desc="default"),
         OptimizerInput(params=None, kwargs={"lr": 0.01}, desc="non-default lr"),
+        OptimizerInput(
+            params=None, kwargs={"lr": torch.tensor(0.01)}, desc="tensor lr"
+        ),
```

Comment:
Not sure how to properly skip the test for this assert:

https://github.com/pytorch/pytorch/blob/8bfd9e98154b83fc6bb0d4407e103d0eb65331ae/torch/optim/adam.py#L466-L469

It is presenting in the CI failure

Reply:
Ah, the way we avoid this is we only have a config for Tensor LR when capturable is defined.

Reply:
Hmm, I think MPS does not support capturable:

https://github.com/pytorch/pytorch/blob/90f82426b96597ca3705c56ee7a58658eb61b69b/torch/optim/optimizer.py#L219-L226

The inputs in `cuda_supported_configs` all have capturable enabled.

Reply:
For now we do not test tensor_lr for Adam & AdamW if the device is not cuda. (Note that we do test tensor_lr for SGD).

Reply:
@janeyx99 what do you think?

Reply:
We can add a mps_supported_configs and add the tensor lr config only if the device is MPS.

Reply:
done :)